node/task.ts

```diff
@@ -1775,6 +1803,218 @@ export class CodeCoverageEnabler {
+    command("task.addattachment", { "type": type, "name": name }, path);
+}
+
+/**
+ * Set an endpoint field with given value.
+ * Value updated will be retained in the endpoint for 
+ * the subsequent tasks that execute within the same job.
+ * 
+ * @param id      Endpoint id.
+ * @param field   FieldType enum of AuthParameter, DataParameter or Url.
+ * @param key     Key.
+ * @param value   Value for key or url.
+ * @returns       void
+ */
+export function setEndpoint(id: string, field: FieldType, key: string, value: string) {
+    command("task.setendpoint", { "id": id, "field": field, "key": key }, value);
+}
+
+/**
+ * Set progress and current operation for current task.
+ * 
+ * @param percent           Percentage of completion.
+ * @param currentOperation  Current pperation. 
+ * @returns                 void
+ */
+export function setProgress(percent: number, currentOperation: string) {
+    command("task.setprogress", { "value": `${percent}` }, currentOperation);
+}
+
+/**
+ * Indicates whether to write the logging command directly to the host or to the output pipeline.
+ *
+ * @param id            Timeline record Guid.
+ * @param parentId      Parent timeline record Guid.
+ * @param recordType    Record type.
+ * @param recordName    Record name.
+ * @param order         Order of timeline record.
+ * @param startTime     Start time.
+ * @param finishTime    End time.
+ * @param progress      Percentage of completion.
+ * @param state         TaskState enum of Unknown, Initialized, InProgress or Completed.
+ * @param result        TaskResult enum of Succeeded, SucceededWithIssues, Failed, Cancelled or Skipped.
+ * @param message       current operation
+ * @returns             void
+ */
+export function logDetail(id: string, parentId: string, recordType: string,
+    recordName: string, order: number, startTime: string, finishTime: string,
+    progress: number, state: TaskState, result: TaskResult, message: string) {
+    const properties = {
+        "id": id,
+        "parentid": parentId,
+        "type": recordType,
+        "name": recordName,
+        "order": order.toString(),
+        "starttime": startTime,
+        "finishtime": finishTime,
+        "progress": progress.toString(),
+        "state": TaskState[state],
+        "result": TaskResult[result]
+    };
+
+    command("task.logdetail", properties, message);
+}
+
+/**
+ * Log error or warning issue to timeline record of current task.
+ *
+ * @param type          IssueType enum of Error or Warning.
+ * @param sourcePath    Source file location.
+ * @param lineNumber    Line number.
+ * @param columnNumber  Column number.
+ * @param code          Error or warning code.
+ * @param message       Error or warning message.
+ * @returns             void
+ */
+export function logIssue(type: IssueType, sourcePath: string, lineNumber: number,
+    columnNumber: number, errorCode: string, message: string) {
```

Comment:
Most of these should be optional - https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md

Reply:
Can you check if the new changes are as expected?

Reply:
Likewise, this can just be:
```
export function logIssue(type: IssueType, message: string, sourcePath?: string, lineNumber?: number,
    columnNumber?: number, errorCode?: string) {
    const properties = {
        "type": type,
        "code": errorCode,
        "sourcepath": sourcePath,
        "linenumber": lineNumber,
        "columnnumber": columnNumber
    };

    command("task.logissue", properties, message);
}
```